src/ngraph/op/reverse.cpp

```diff
@@ -69,3 +70,132 @@ void op::Reverse::generate_adjoints(autodiff::Adjoints& adjoints, const NodeVect
+                         const Output<Node>& reversed_axes,
+                         const std::string& mode)
+    : Op({data, reversed_axes})
+    , m_mode{mode_from_string(mode)}
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v1::Reverse::Reverse(const Output<Node>& data,
+                         const Output<Node>& reversed_axes,
+                         const Mode mode)
+    : Op({data, reversed_axes})
+    , m_mode{mode}
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v1::Reverse::validate_and_infer_types()
+{
+    if (m_mode == Mode::MASK)
+    {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(1) == element::boolean,
+                              "In 'mask' mode the second input must contain boolean values.");
+    }
+
+    const auto input_shape = get_input_partial_shape(0);
+    const auto input_rank = input_shape.rank();
+
+    const auto rev_axes_shape = get_input_partial_shape(1);
+    const auto rev_axes_rank = rev_axes_shape.rank();
+
+    if (rev_axes_rank.is_static())
+    {
+        NODE_VALIDATION_CHECK(this,
+                              static_cast<size_t>(rev_axes_rank) == 1,
+                              "The reversed_axes input must be a 1D tensor (got ",
+                              static_cast<size_t>(rev_axes_rank),
+                              ").");
+
+        if (m_mode == Mode::MASK)
+        {
+            NODE_VALIDATION_CHECK(
+                this,
+                static_cast<size_t>(rev_axes_rank) == 1,
+                "The reversed_axes input rank needs to be equal to 1 in 'mask' mode.");
+
+            if (input_rank.is_static() && rev_axes_shape[0].is_static())
+            {
+                const auto rev_axes_mask_elems_count = static_cast<size_t>(rev_axes_shape[0]);
+                NODE_VALIDATION_CHECK(this,
+                                      rev_axes_mask_elems_count == static_cast<size_t>(input_rank),
+                                      "The number of elements in the reversed_axes tensor (",
+                                      rev_axes_mask_elems_count,
+                                      ") must match the input data tensor rank (",
+                                      static_cast<size_t>(input_rank),
+                                      ") in 'mask' mode.");
+            }
+        }
+    }
+
+    if (input_rank.is_static())
+    {
+        const auto rank = static_cast<size_t>(input_rank);
+        const auto rev_axes_node = input_value(1).get_node_shared_ptr();
+
+        if (rev_axes_node->is_constant())
+        {
+            const auto rev_axes_constant = dynamic_pointer_cast<op::Constant>(rev_axes_node);
+
+            if (m_mode == Mode::INDEX)
+            {
+                const AxisSet rev_axes = rev_axes_constant->get_axis_set_val();
+
+                NODE_VALIDATION_CHECK(this,
+                                      rev_axes.size() < rank,
```

Comment:
If all axes are set to be reversed, which is a valid case, this check fails. Should be `<=`.